src/main/scala/TraversalAlgHom.scala

```diff
@@ -0,0 +1,54 @@
+package org.hablapps.statelesser
+
+import scalaz._, Scalaz._
+import shapeless._
+
+trait TraversalAlgHom[Alg[_[_], _], P[_], A] {
+  type Q[_]
+  val alg: Alg[Q, A]
+  val hom: Q ~> ListT[P, ?]
+
+  def apply[B](f: Alg[Q, A] => Q[B]): P[List[B]] =
+    hom(f(alg)).run
+
+  def fold[B](f: Alg[Q, A] => Q[B])(implicit 
+      F: Functor[P],
+      M: Monoid[B]): P[B] =
+    apply(f).map(_.suml)
+
+  def composeLens[Alg2[_[_], _], B](
+      ln: LensAlgHom[Alg2, Q, B]): TraversalAlgHom.Aux[Alg2, P, ln.Q, B] =
+    TraversalAlgHom[Alg2, P, ln.Q, B](ln.alg, hom compose ln.hom)
+}
+
+object TraversalAlgHom {
+
+  type Aux[Alg[_[_], _], P[_], Q2[_], A] = 
+    TraversalAlgHom[Alg, P, A] { type Q[x] = Q2[x] }
+
+  def apply[Alg[_[_], _], P[_], Q2[_], A](
+      alg2: Alg[Q2, A],
+      hom2: Q2 ~> ListT[P, ?]): Aux[Alg, P, Q2, A] =
+    new TraversalAlgHom[Alg, P, A] {
+      type Q[x] = Q2[x]
+      val alg = alg2
+      val hom = hom2
+    }
+
+  implicit def genTraversalAlgHom[H, T <: HList, Alg[_[_], _], S, A](implicit 
+      ge: GetEvidence[HNil, Alg[State[A, ?], A]],
+      fl: MkFieldLens.Aux[S, H, List[A]])
+      : GetEvidence[H :: T, TraversalAlgHom[Alg, State[S, ?], A]] =
+    GetEvidence(TraversalAlgHom(ge(), fl()))
```

Comment:
Ideally, the type of `fl` should be `MkFieldLens.Aux[S, H, F[A]]` where `F[_]: Traverse` (notice the fixed `List`). However, if we apply such change, the compiler doesn't detect this evidence generator. Any workaround to prevent this situation?

Reply:
The `Foldable` type class should be enough, if we just want something that can be converted to a list. `Traversable` gives you extra power: https://wiki.haskell.org/Foldable_and_Traversable

Reply:
`Foldable` isn't a `Functor` and I need the `F` to be mappable, unless there is a more clever implementation...